File: AoC2019Tests/Day01Test.fs
module Day01Test

open AoC2019

open NUnit.Framework

[<Test>]
let Part1 () =
    let result = Day01.SolvePart1
    Assert.AreEqual(result, "3331849")


[<Test>]
let Part2 () = 
    let result = Day01.SolvePart2
    Assert.AreEqual(result, "4994898")

        
[<TestCase(12, 2)>]
[<TestCase(14, 2)>]
[<TestCase(1969, 654)>]
[<TestCase(100756, 33583)>]
let GetFuelTest(input : int,  expected : int) =
    let actual = Day01.GetFuel(input)
    Assert.AreEqual(expected, actual)


[<TestCase(14, 2)>]
[<TestCase(1969, 966)>]
[<TestCase(100756, 50346)>]
let GetFuelRecursiveTest(input: int, expected: int) = 
    let actual = Day01.GetFuelRecursive(input)
    Assert.AreEqual(expected, actual)

File: AoC2019Tests/Day01Test.cs
namespace AoC2019Tests;

public class Day01Test
{
	Day01 day01;

	[SetUp]
	public void Setup( )
	{
		day01 = new Day01("day01");
	}
        
	[TestCase(12,2)] 
	[TestCase(14,2)] 
	[TestCase(1969,654)] 
	[TestCase(100756,33583)]
	public void GetFuelPerModule(int input, int expected)
	{
		var actual = Day01.GetFuelPerModule(input);
		Assert.AreEqual(expected, actual);
	}

	[Test]
	public void Part1( )
	{
		var actual = day01.SolvePart1( ).Result;
		Assert.AreEqual("3331849", actual);
	}

	[Test]
	public void Part2( )
	{
		var actual = day01.SolvePart2( ).Result;
		Assert.AreEqual("4994898", actual);
	}
}